content/blog/azure-pipelines-scaler.md

```diff
@@ -0,0 +1,224 @@
++++
+title = "Autoscaling Azure Pipelines agents with KEDA"
+date = 2021-04-28
+author = "Troy Denorme"
++++
+
+With the addition of Azure Piplines support in KEDA, it is now possible to autoscale your Azure Pipelines agents based on the agent pool queue length.
+Self-hosted Azure Pipelines agents are the perfect workload for this scaler. By autoscaling the agents you can create a scalable CI/CD environment.
+
+> The number of concurrent pipelines you can run is limited by your [parallel jobs](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents#parallel-jobs). KEDA will autoscale to the maximum defined in the ScaledObject and does not limit itself to the parallel jobs count defined for the Azure DevOps organization.
+
+## Azure Pipelines agents
+
+Azure Pipelines jobs can run on different kinds of agents ([docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents)). But if you want full control, you are going to use self-hosted agents. 
+The agents can run on Linux, macOS or Windows machines but also in a container. To scale the agents with KEDA you can run self-hosted agents on Kubernetes.
+
+You can run the agents as a `Deployment` or a `Job` in Kubernetes and scale them accordingly with a `ScaledObject` or a `ScaledJob`.
+
+## Deploy a self-hosted agent on Kubernetes as a Deployment
+
+### Create the container image
+
+To create a basic Azure Pipelines agent image you can follow the instructions from the offical docs. [Run a self-hosted agent in Docker](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/docker#linux).
+
+### Deploy on Kubernetes
+
+You can deploy the agent as a Kubernetes deployment and use KEDA to autoscale the deployment.
+The following is an example manifest for the deployment:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: azdevops-deployment
+  labels:
+    app: azdevops-agent
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: azdevops-agent
+  template:
+    metadata:
+      labels:
+        app: azdevops-agent
+    spec:
+      containers:
+      - name: azdevops-agent
+        image: <azdevops-image>
+        env:
+          - name: AZP_URL
+            valueFrom:
+              secretKeyRef:
+                name: azdevops
+                key: AZP_URL
```

Comment:
Might be good to add an example Secret for a full reference?

Reply:
I have changed the example a bit and included the secret as well.